pkg/kubelet/cri/streaming/portforward/httpstream.go

```diff
@@ -252,11 +261,24 @@ func (h *httpStreamHandler) portForward(p *httpStreamPair) {
 	err := h.forwarder.PortForward(ctx, h.pod, h.uid, int32(port), p.dataStream)
 	klog.V(5).InfoS("Connection request done invoking forwarder.PortForward for port", "connection", h.conn, "request", p.requestID, "port", portString)
 
-	if err != nil {
-		msg := fmt.Errorf("error forwarding port %d to pod %s, uid %v: %v", port, h.pod, h.uid, err)
-		utilruntime.HandleError(msg)
-		fmt.Fprint(p.errorStream, msg.Error())
+	// happy path, we have successfully completed forwarding task
+	if err == nil {
+		return
 	}
+
+	// forwarding in one direction has been completed.
+	// other direction: we got RST, so we should reset data && err stream
+	if errors.Is(err, syscall.EPIPE) {
```

Comment:
one question, if we got reset don't we get a `ECONNRESET` error?

Reply:
oh, I see, so the problem is when we write to a closed pipe, because the target container no longer runs anything

Reply:
the container not, but the process listening on that port, right?

Reply:
yes~

Reply:
> one question, if we got reset don't we get a ECONNRESET error?

xref: https://github.com/kubernetes/kubernetes/issues/74551#issuecomment-1550655795

Reply:
yeah, but think that this code is in different places, so the same code in containerd and kubelet will handle different errors. If the connection has been closed by the RST and one side tries to write, it will get an EPIPE error.
I don't know how the `ECONNRESET` is surfaced though, when it tries to read I imagine

Reply:
ECONNRESET is another story, ECONNRESET is handled here because of brianpursley's suggestion.
xref: https://github.com/kubernetes/kubernetes/pull/117493#discussion_r1191478301